src/test/java/StringManipulationTest.java

```diff
@@ -88,28 +118,53 @@ public void testGeSubStrings1() {
     }
 
     @Test
-    public void testGeSubStrings2() {
-        fail("Not yet implemented");
+    public void testGetSubStrings2_ThrowIllegalArgumentException_startWordLessThanZero() {
+        manipulatedstring.setString("I am a String!");
+
+        assertThrows(IllegalArgumentException.class,
+                ()-> {
+                    String [] sStings = manipulatedstring.getSubStrings(-1, 4);
+                });
     }
     @Test
-    public void testGeSubStrings3() {
-        fail("Not yet implemented");
+    public void testGetSubStrings3_ThrowIllegalArgumentException_endWordLessThanZero() {
+        manipulatedstring.setString("I am a String!");
+
+        assertThrows(IllegalArgumentException.class,
+                ()-> {
+                    String [] sStings = manipulatedstring.getSubStrings(1, -1);
+                });
     }
     @Test
-    public void testGeSubStrings4() {
-        fail("Not yet implemented");
+    public void testGetSubStrings4_ThrowIllegalArgumentException_endWordLessThanstartWord() {
+        manipulatedstring.setString("I am a String!");
+
+        assertThrows(IllegalArgumentException.class,
+                ()-> {
+                    String [] sStings = manipulatedstring.getSubStrings(2, 1);
+                });
     }
     @Test
-    public void testGeSubStrings5() {
-        fail("Not yet implemented");
+    public void testGetSubStrings5_ThrowIndexOutOfBoundsException_endWordGreaterThanNumberOfWords() {
+        manipulatedstring.setString("I am a String!");
+
+        assertThrows(IndexOutOfBoundsException.class,
+                ()-> {
+                    String [] sStings = manipulatedstring.getSubStrings(1, 5);
+                });
     }
     @Test
-    public void testGeSubStrings6() {
-        fail("Not yet implemented");
+    public void testGetSubStrings6_EmptyString() {
+        manipulatedstring.setString("");
+
+        assertThrows(IndexOutOfBoundsException.class,
+                ()-> {
+                    String [] sStings = manipulatedstring.getSubStrings(1, 5);
+                });
     }
 
```

Comment:
All test cases use the same string, consider adding tests that deal with numbers/symbols/whitespace.